airflow/sensors/time_sensor.py

```diff
@@ -72,13 +72,13 @@ def __init__(self, *, target_time, **kwargs):
 
         self.target_datetime = timezone.convert_to_utc(aware_time)
 
-    def execute(self, context: Context):
+    def execute(self, context: Context) -> None:
```

Comment:
Maybe in this case more suitable is [NoReturn](https://docs.python.org/3/library/typing.html#typing.NoReturn)?

Reply:
Agree. NoReturn is better here

Reply:
Good call, TIL. I've removed it from this PR and will do it in a follow up.

Reply:
#39742